src/App.tsx

```diff
@@ -1,32 +1,46 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
+import { useEffect, useState } from 'react'
 import './App.css'
+import { CartPage } from './domain/cart/CartPage'
+import { BooksPage } from './domain/books/BooksPage'
+import Navbar from './components/navbar/navbar'
+import { getBooks, useAppDispatch, useAppSelector } from './domain/books/booksSlice'
+import { Book } from './domain/books/types'
+import { addToCart,removeFromCart} from './domain/cart/cartSlice'
+import { CartItem, CartState } from './domain/cart/types'
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [route, setRoute] = useState<number>(0)
+  const dispatch = useAppDispatch()
+  const { books,loading } = useAppSelector(state => state.books)
+  const { cartItems } = useAppSelector(state => state.cart)
+
+useEffect(() => {
+  dispatch(getBooks())
+}, [])
+
+  const changeRoute = (x: number): void => {
+    setRoute(x)
+  }
+  const handleAddToCart = (item:Book): void=>{
+    console.log(item)
+    dispatch(addToCart(item))
+  }
+  const handleRemoveCartItem = (item:CartItem):void=>{
+    dispatch(removeFromCart(item))
+  }
+  const getCartTotalCurrency = ({cartItems}: CartState):number=>{
+    let total :number= 0
+    cartItems.forEach((item:CartItem)=> total +=item?.price)
+    return total;
+  }
 
   return (
-    <div className="App">
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src="/vite.svg" className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://reactjs.org" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+    <div className="main">
+       <Navbar changeRoute={changeRoute}/>
+      <hr></hr>
+      {
+        route === 1 ? <CartPage cartItems={cartItems} getCartTotalCurrency={getCartTotalCurrency}handleRemoveCartItem={handleRemoveCartItem} /> : <BooksPage books={books} loading={loading} handleAddToCart={handleAddToCart} />
```

Comment:
There are a couple of things
1. Declare the routes in a separate file.
2. No need to pass these props. Let each Page component handle it.